File: frontend/types/payment.ts
export interface CreateChargeResponse {
  billing_id: string;
  pix_qr_code: string;
  pix_copy_paste: string;
  expires_at: string;
  expires_in: number;
  amount_cents: number;
}

export interface PaymentStreamEvent {
  event?: "payment_confirmed" | "payment_expired" | string;
  analysis_credits?: number;
}

File: backend/app/schemas/payment.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class CreateChargeResponse(BaseModel):
    billing_id: str
    pix_qr_code: str
    pix_copy_paste: str
    expires_at: datetime
    expires_in: int
    amount_cents: int


class WebhookResponse(BaseModel):
    received: bool
    status: str
